docker/models/services.py

```diff
@@ -0,0 +1,89 @@
+import inspect
+from docker.types import TaskTemplate, ContainerSpec
+from .resource import Model, Collection
+
+
+class Service(Model):
+    """A service."""
+    id_attribute = 'ID'
+
+    @property
+    def name(self):
+        """The service's name."""
+        return self.attrs['Spec']['Name']
+
+    @property
+    def version(self):
+        """The version number of this service."""
+        return self.attrs.get('Version').get('Index')
+
+    def remove(self):
+        """Remove this service."""
+        return self.client.api.remove_service(self.id)
+
+    def tasks(self):
+        """List the tasks in this service."""
+        return self.client.api.tasks(filters={'service': self.id})
+
+    def update(self, **kwargs):
+        """Update a service's configuration."""
+        create_kwargs, spec_kwargs = _get_create_and_spec_kwargs(kwargs)
+        # Image is required, so if it hasn't been set, use current image
+        if 'image' not in spec_kwargs:
+            spec = self.attrs['Spec']['TaskTemplate']['ContainerSpec']
+            spec_kwargs['image'] = spec['Image']
+        return self.client.api.update_service(
+            self.id,
+            self.version,
+            TaskTemplate(ContainerSpec(**spec_kwargs)),
+            **create_kwargs
+        )
+
+
+class ServiceCollection(Collection):
+    """Services on the Docker server."""
+    model = Service
+
+    def create(self, **kwargs):
+        """Create a service."""
+        create_kwargs, spec_kwargs = _get_create_and_spec_kwargs(kwargs)
+        service_id = self.client.api.create_service(
+            TaskTemplate(ContainerSpec(**spec_kwargs)),
+            **create_kwargs
+        )
```

Comment:
We need to find some shortcut to allow users to do the most common things with a flat list of args, because constructing objects like this is a nightmare for the user.

I disagree that it is inherently complicated. It works fine in the CLI, and this is what the high-level API should be replicating.

Perhaps `client.services.create()` should have these arguments:

```
$ docker service create --help                                                      !65

Usage:	docker service create [OPTIONS] IMAGE [COMMAND] [ARG...]

Create a new service

Options:
      --constraint value               Placement constraints (default [])
      --container-label value          Container labels (default [])
      --endpoint-mode string           Endpoint mode (vip or dnsrr)
  -e, --env value                      Set environment variables (default [])
      --help                           Print usage
  -l, --label value                    Service labels (default [])
      --limit-cpu value                Limit CPUs (default 0.000)
      --limit-memory value             Limit Memory (default 0 B)
      --log-driver string              Logging driver for service
      --log-opt value                  Logging driver options (default [])
      --mode string                    Service mode (replicated or global) (default "replicated")
      --mount value                    Attach a mount to the service
      --name string                    Service name
      --network value                  Network attachments (default [])
  -p, --publish value                  Publish a port as a node port (default [])
      --replicas value                 Number of tasks (default none)
      --reserve-cpu value              Reserve CPUs (default 0.000)
      --reserve-memory value           Reserve Memory (default 0 B)
      --restart-condition string       Restart when condition is met (none, on-failure, or any)
      --restart-delay value            Delay between restart attempts (default none)
      --restart-max-attempts value     Maximum number of restarts before giving up (default none)
      --restart-window value           Window used to evaluate the restart policy (default none)
      --stop-grace-period value        Time to wait before force killing a container (default none)
      --update-delay duration          Delay between updates
      --update-failure-action string   Action on update failure (pause|continue) (default "pause")
      --update-parallelism uint        Maximum number of tasks updated simultaneously (0 to update all at once) (default 1)
  -u, --user string                    Username or UID
      --with-registry-auth             Send registry authentication details to swarm agents
  -w, --workdir string                 Working directory inside the container
```

If you want to do anything else more complex, you can use the low-level `APIClient.create_service()` method, or pass in a `TaskTemplate` or something.